pkgs/development/libraries/dbus/default.nix

```diff
@@ -1,92 +1,115 @@
-{ stdenv, lib, fetchurl, pkgconfig, expat, systemd
-, libX11 ? null, libICE ? null, libSM ? null, x11Support ? (stdenv.isLinux || stdenv.isDarwin) }:
+{ stdenv
+, lib
+, fetchurl
+, pkgconfig
+, expat
+, systemd
+, libX11 ? null
+, libICE ? null
+, libSM ? null
+, x11Support ? (stdenv.isLinux || stdenv.isDarwin)
+, dbus
+}:
 
 assert x11Support -> libX11 != null
-                  && libICE != null
-                  && libSM != null;
+  && libICE != null
+  && libSM != null
+;
 
 let
+
+dbus = stdenv.mkDerivation rec {
```

Comment:
This just makes it ```let dbus = ... in dbus``` right? Can we just make it a simple ```stdenv.mkDerivation```.

Reply:
We want to reference `dbus` in `passthru` and this is the way to do that.
I just changed it from `self` and didn't indent it because it's not that easy to read when it's just let into it's self.

Reply:
It seems to work even with the dbus passed by the `callPackage` so maybe we can get rid of self after all.

Reply:
So no changes needed?

Reply:
```patch
diff --git a/pkgs/development/libraries/dbus/default.nix b/pkgs/development/libraries/dbus/default.nix
index 34f1a9c3af5..c962f50d3b8 100644
--- a/pkgs/development/libraries/dbus/default.nix
+++ b/pkgs/development/libraries/dbus/default.nix
@@ -15,9 +15,7 @@ assert
   x11Support ->
     libX11 != null && libICE != null && libSM != null;
 
-let
-
-dbus = stdenv.mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "dbus";
   version = "1.12.16";
 
@@ -109,6 +107,4 @@ dbus = stdenv.mkDerivation rec {
     license = licenses.gpl2Plus; # most is also under AFL-2.1
     platforms = platforms.unix;
   };
-};
-
-in dbus
+}

```

Reply:
There's probably a few expressions that could benefit from this change.